packages/react-component-library/src/components/DatePicker/CalendarTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { IconChevronLeft, IconChevronRight } from '@royalnavy/icon-library'
+import { setMonth as changeMonth, setYear as changeYear } from 'date-fns/fp'
+import React, { useState, Dispatch, SetStateAction } from 'react'
+
+import { BUTTON_VARIANT } from '../Button'
+import { CALENDAR_TABLE_VIEW, type CalendarTableView } from './constants'
+import { DATEPICKER_ACTION, type DatePickerAction } from './types'
+import {
+  StyledButtonArrows,
+  StyledCalendarContainer,
+  StyledCalendarTable,
+  StyledCalendarTiles,
+  StyledNavigation,
+  StyledYearsTitle,
+} from './partials/StyledCalendarNavigation'
+
+interface CalendarTableProps {
+  calendarTable: CalendarTableView
+  dispatch: Dispatch<DatePickerAction>
+  month: Date
+  setCalendarTable: Dispatch<SetStateAction<CalendarTableView>>
+}
+
+const MONTHS = [
+  'JAN',
+  'FEB',
+  'MAR',
+  'APR',
+  'MAY',
+  'JUN',
+  'JUL',
+  'AUG',
+  'SEP',
+  'OCT',
+  'NOV',
+  'DEC',
+]
+
+const getDecade = (year: number) =>
+  Array.from({ length: 12 }, (_, i) => Math.floor(year / 10) * 10 + i - 1)
```

Comment:
Maybe abstracting some of this `Math.floor(year / 10) * 10 + i` to a variable name(s) will make it a little more self documenting.

Reply:
Some other things that might help:

- using [range()](https://lodash.com/docs/4.17.15#range) from lodash
- renaming the function to `getDecadeYears`